### ISSUE TITLE
Test the Groovy cron tasks

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#594]
* Create some dummy Groovy scripts (perhaps one that prints the date and another -- the time)
* Create cron tasks for these and see how it behaves

Check out the [Cron-Tasks](https://github.com/strongbox/strongbox/wiki/Cron-Tasks) for more details. This task relates to #521.

**Comment by rmereddy** [comment]
Hi,
Is this issue still open?

Thanks,
Rohit

**Comment by carlspring** [comment]
Hi,

Yes! Would you like to help? You can join our chat on Gitter (
http://gitter.im/strongbox/strongbox/).

See you there!

Martin

**Comment by andreybleme** [comment]
Hi @carlspring can you assign this task to me? I'll work on it. I'm joining the gitter to gather ideas about how we can better test the crons.

**Comment by carlspring** [comment]
@andreybleme ,

How's it going? Any issues, or updates?

**Comment by bamarch** [comment]
I'll test out a simple/dummy Groovy script and add it to the wiki documentation for Groovy cron tasks

**Comment by bamarch** [comment]
I think this is a duplicate of #521

**Comment by bamarch** [comment]
I took a look at CronTaskConfigurationControllerTest, which demonstrates:

```
CronTaskConfigurationDto configuration = new CronTaskConfigurationDto();
configuration.setOneTimeExecution(true);
configuration.setName(name);
configuration.addProperty("cronExpression", cronExpression);
```

This instantiates some config for a groovy cron task (the name, interval it runs at etc...)

Then when the app is running, it seems you hit the correct endpoint with the groovy script body and pass in cronName parameter

```
PUT - http://localhost:48080/configuration/crontasks/crontask/groovy/?cronName=MyCron
```

where MyCron would match configuration.setName(name).

So one question

- Where would be the appropriate place that all the groovy cron configuration should be defined?

**Comment by fuss86** [comment]
> PUT - http://localhost:48080/configuration/crontasks/crontask/groovy/?cronName=MyCron

it should be

`PUT - http://localhost:48080/api/configuration/crontasks/cron/groovy?cronName=MyCron`

**Comment by carlspring** [comment]
Yeah, you will have to update documentation for that, @bamarch... (Thanks! @fuss86!)

**Comment by fuss86** [comment]
`Where would be the appropriate place that all the groovy cron configuration should be defined?`

are you asking for 

`String path = ConfigurationResourceResolver.getVaultDirectory() + "/etc/conf/cron/groovy";` 

?

**Comment by bamarch** [comment]
ConfigurationResourceResolver.getVaultDirectory() + "/etc/conf/cron/groovy" is where the script is stored after hitting the endpoint

`PUT - http://localhost:48080/api/configuration/crontasks/cron/groovy?cronName=MyCron` 

What I meant to ask for is... when you upload a script it only contains the body. Where do you first define the name, cron expression etc...? 

But I took a look at `CronTaskController` which answered my questions, I think.

You have to create a xml element `cron-task-configuration`, for instance `strongbox-cron/strongbox-cron-api/src/main/resources/etc/conf/strongbox-cron-tasks.xml`.

**Comment by bamarch** [comment]
nvm updated docker-compose.yml to use spring boot

**Comment by bamarch** [comment]
the tests in CronTaskConfigurationControllerTest all pass, and reviewing this class I got a list of the correct relevant paths for uploading groovy scripts, doing the cron configuration etc...

the issue is that I get this error when trying to make this requests to an instance of the application

`| org.springframework.web.servlet.PageNotFound       | No mapping found for HTTP request with URI [/api/configuration/crontasks/groovy/names] in DispatcherServlet with name 'dispatcherServlet'`

is this normal...i.e. todo in spring, or is this something that needs fixing somewhere? :)

**Comment by fuss86** [comment]
It looks like `strongbox-cron-rest` is not in the classpath, thanks for spotting this !

Will you be able to add necessary dependency to make it work ?

**Comment by bamarch** [comment]
Yep I’ll bundle it with the change to remove the outdated readme.md stuff
